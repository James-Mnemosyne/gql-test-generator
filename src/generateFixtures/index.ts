import { FixtureNode, FixtureNodeVariation } from '../types';
import { generateFixturesForEnumFixtureNode } from './generateFixturesForEnumFixtureNode';
import { generateFixturesForFunctionFixtureNode } from './generateFixturesForFunctionFixtureNode';
import { generateFixturesForObjectFixtureNode } from './generateFixturesForObjectFixtureNode';
import { generateFixturesForScalarNode } from './generateFixturesForScalarNode';

function compress(fixtures: any[], maxFixtures: number): any[] {
  if (fixtures.length <= maxFixtures) {
    return fixtures;
  }

  const results: any[] = [];
  for (let index = 0; index < fixtures.length; index += (fixtures.length / (maxFixtures - 1))) {
    const newValue = fixtures[Math.floor(index)];
    results.push(newValue);
  }
  return results;
}

export function generateFixtures(node: FixtureNode, fixtureNodes: Record<string, FixtureNode>, maxFixtures: number): any[] {
  switch (node.variation) {
    case FixtureNodeVariation.ScalarFixtureNode:
      const scalarResults = generateFixturesForScalarNode(node, fixtureNodes, maxFixtures);
      return compress(scalarResults, maxFixtures);
    case FixtureNodeVariation.FunctionFixtureNode:
      const functionResults = generateFixturesForFunctionFixtureNode(node, fixtureNodes, maxFixtures);
      return compress(functionResults, maxFixtures);
    case FixtureNodeVariation.ObjectFixtureNode:
      const objectResults = generateFixturesForObjectFixtureNode(node, fixtureNodes, maxFixtures);
      return compress(objectResults, maxFixtures);
    case FixtureNodeVariation.EnumFixtureNode:
      const enumResults = generateFixturesForEnumFixtureNode(node);
      return compress(enumResults, maxFixtures);
    default:
      throw new Error(`Unsupported generic fixture type generation for ${node.variation}.`);
  }
}
