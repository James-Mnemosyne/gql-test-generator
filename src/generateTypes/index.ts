import * as graphql from 'graphql';
import { loadSchema } from '@graphql-tools/load';
import { GraphQLFileLoader } from '@graphql-tools/graphql-file-loader';

import * as fs from 'fs';
import { FixtureNode, Config } from '../types';
import { getNewResolve } from '../getSchemaVisitor';
import { generateFixtures } from '../generateFixtures';
import { IsObjectFixtureNodeOrUndefined } from '../utils/filters';
import { getTests } from '../getTests';
import { getDocumentNodeFromSchema, memoize1 } from '@graphql-tools/utils';
import { GraphQLSchema } from 'graphql';

const getCachedDocumentNodeFromSchema = memoize1(getDocumentNodeFromSchema);

function getFileName(config: Config) {
  const initialFileNameSegments = config.resolverFilePath.split('/');
  const fileName = initialFileNameSegments.pop();
  const fileNameSegments = fileName.split('.');
  const extension = fileNameSegments.pop();
  const newFileName = [...fileNameSegments, 'spec', extension].join('.');
  const newPath = [...initialFileNameSegments, newFileName].join('/');
  return newPath;
}

export async function generateTypes(config: Config): Promise<string> {
  const schema: GraphQLSchema = await loadSchema(config.schemaFilePath, { loaders: [new GraphQLFileLoader()] });
  const ast = getCachedDocumentNodeFromSchema(schema);
  const fixtureNodes: Record<string, FixtureNode> = {};
  const newResolve = getNewResolve(fixtureNodes);

  graphql.visit(ast, { leave: newResolve });

  const queryNode = fixtureNodes?.Query;
  const mutationNode = fixtureNodes?.Mutation;
  if (!IsObjectFixtureNodeOrUndefined(queryNode)) {
    throw new Error('Query is not object or undefined.');
  }
  if (!IsObjectFixtureNodeOrUndefined(mutationNode)) {
    throw new Error('Mutation is not object or undefined.');
  }
  const baseNode =
    queryNode?.fields?.find((field) => field.name === config.resolverName) ??
    mutationNode?.fields?.find((field) => field.name === config.resolverName);
  const fixtures = generateFixtures(baseNode, fixtureNodes, config.maxFixtures ?? 100);

  const result = getTests(fixtures, config);
  console.log(result);

  const fileName = getFileName(config);
  fs.writeFileSync(fileName, result);

  return result;
}
