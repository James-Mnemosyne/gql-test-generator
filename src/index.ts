#!/usr/bin/env node

import { generateTypes } from './generateTypes';
import { getConfig } from './getConfig';

async function run() {
  const config = getConfig();
  await generateTypes(config);
}

run()
  .then(() => {
    process.exit(0);
  })
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
