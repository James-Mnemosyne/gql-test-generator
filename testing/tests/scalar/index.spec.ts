import { generateTypes } from '../../../src/index';
import * as fs from 'fs';

describe('Query', () => {
  it('with only basic scalars', async () => {
    fs.writeFileSync('./testing/results/scalar/index.spec.ts', '');

    const result = await generateTypes({
      resolverName: 'getSomething',
      schemaFilePath: './testing/fixtures/scalar/schema.graphql',
      resolverFilePath: './testing/results/scalar/index.ts',
      typeFilePath: './testing/fixtures/types.generated.ts',
    });

    expect(result).toMatchSnapshot();
  });
});
