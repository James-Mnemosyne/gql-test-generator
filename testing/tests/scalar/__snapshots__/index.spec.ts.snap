// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query with only basic scalars 1`] = `
"
import { GraphQLResolveInfo } from 'graphql';
import { getSomething } from './testing/results/scalar/index.ts';


// We never actually use the info object.
export const mockGraphQLResolveInfo = {
  fieldName: '',
  fieldNodes: [],
  returnType: undefined,
  parentType: undefined,
  path: undefined,
  schema: undefined,
  fragments: undefined,
  rootValue: undefined,
  operation: undefined,
  variableValues: {},
  cacheControl: undefined,
} as unknown as GraphQLResolveInfo;

describe('getSomething smoke tests', () => {
  beforeAll(() => {
    // Do something here.
  });

  beforeEach(() => {
    // Do something here.
  });


  it('case0', async () => {
    const params = {"str":"","addressEmail":"","url":"","numFloat":-1000.5,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case1', async () => {
    const params = {"str":"","addressEmail":"","url":"","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case2', async () => {
    const params = {"str":"","addressEmail":"","url":"random string","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case3', async () => {
    const params = {"str":"","addressEmail":"","url":"http://www.google.com","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case4', async () => {
    const params = {"str":"","addressEmail":"","url":"http://www.google.com","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case5', async () => {
    const params = {"str":"","addressEmail":"random string","url":"","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case6', async () => {
    const params = {"str":"","addressEmail":"random string","url":"random string","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case7', async () => {
    const params = {"str":"","addressEmail":"random string","url":"http://www.google.com","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case8', async () => {
    const params = {"str":"","addressEmail":"random string","url":"http://www.google.com","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case9', async () => {
    const params = {"str":"","addressEmail":"james@bob.edu","url":"","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case10', async () => {
    const params = {"str":"","addressEmail":"james@bob.edu","url":"random string","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case11', async () => {
    const params = {"str":"","addressEmail":"james@bob.edu","url":"random string","numFloat":2000345.7898,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case12', async () => {
    const params = {"str":"","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case13', async () => {
    const params = {"str":"random string","addressEmail":"","url":"","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case14', async () => {
    const params = {"str":"random string","addressEmail":"","url":"random string","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case15', async () => {
    const params = {"str":"random string","addressEmail":"","url":"random string","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case16', async () => {
    const params = {"str":"random string","addressEmail":"","url":"http://www.google.com","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case17', async () => {
    const params = {"str":"random string","addressEmail":"random string","url":"","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case18', async () => {
    const params = {"str":"random string","addressEmail":"random string","url":"random string","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case19', async () => {
    const params = {"str":"random string","addressEmail":"random string","url":"random string","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case20', async () => {
    const params = {"str":"random string","addressEmail":"random string","url":"http://www.google.com","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case21', async () => {
    const params = {"str":"random string","addressEmail":"james@bob.edu","url":"","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case22', async () => {
    const params = {"str":"random string","addressEmail":"james@bob.edu","url":"","numFloat":2000345.7898,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case23', async () => {
    const params = {"str":"random string","addressEmail":"james@bob.edu","url":"random string","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case24', async () => {
    const params = {"str":"random string","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case25', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"","url":"","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case26', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"","url":"","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case27', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"","url":"random string","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case28', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"","url":"http://www.google.com","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case29', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"random string","url":"","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case30', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"random string","url":"","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case31', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"random string","url":"random string","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case32', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"random string","url":"http://www.google.com","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case33', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"random string","url":"http://www.google.com","numFloat":2000345.7898,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case34', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"james@bob.edu","url":"","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case35', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"james@bob.edu","url":"random string","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case36', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case37', async () => {
    const params = {"optStr":null,"str":"","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case38', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"","url":"","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case39', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"","url":"random string","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case40', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"","url":"http://www.google.com","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case41', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"","url":"http://www.google.com","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case42', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"random string","url":"","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case43', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"random string","url":"random string","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case44', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"random string","url":"http://www.google.com","numFloat":-1000.5,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case45', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"random string","url":"http://www.google.com","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case46', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"james@bob.edu","url":"","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case47', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"james@bob.edu","url":"random string","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case48', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"james@bob.edu","url":"random string","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case49', async () => {
    const params = {"optStr":null,"str":"random string","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case50', async () => {
    const params = {"optStr":"","str":"","addressEmail":"","url":"","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case51', async () => {
    const params = {"optStr":"","str":"","addressEmail":"","url":"random string","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case52', async () => {
    const params = {"optStr":"","str":"","addressEmail":"","url":"random string","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case53', async () => {
    const params = {"optStr":"","str":"","addressEmail":"","url":"http://www.google.com","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case54', async () => {
    const params = {"optStr":"","str":"","addressEmail":"random string","url":"","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case55', async () => {
    const params = {"optStr":"","str":"","addressEmail":"random string","url":"random string","numFloat":-1000.5,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case56', async () => {
    const params = {"optStr":"","str":"","addressEmail":"random string","url":"random string","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case57', async () => {
    const params = {"optStr":"","str":"","addressEmail":"random string","url":"http://www.google.com","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case58', async () => {
    const params = {"optStr":"","str":"","addressEmail":"james@bob.edu","url":"","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case59', async () => {
    const params = {"optStr":"","str":"","addressEmail":"james@bob.edu","url":"","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case60', async () => {
    const params = {"optStr":"","str":"","addressEmail":"james@bob.edu","url":"random string","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case61', async () => {
    const params = {"optStr":"","str":"","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case62', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"","url":"","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case63', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"","url":"","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case64', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"","url":"random string","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case65', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"","url":"http://www.google.com","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case66', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"random string","url":"","numFloat":-1000.5,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case67', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"random string","url":"","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case68', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"random string","url":"random string","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case69', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"random string","url":"http://www.google.com","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case70', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"random string","url":"http://www.google.com","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case71', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"james@bob.edu","url":"","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case72', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"james@bob.edu","url":"random string","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case73', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case74', async () => {
    const params = {"optStr":"","str":"random string","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case75', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"","url":"","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case76', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"","url":"random string","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case77', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"","url":"http://www.google.com","numFloat":-1000.5,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case78', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"","url":"http://www.google.com","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case79', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"random string","url":"","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case80', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"random string","url":"random string","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case81', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"random string","url":"random string","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case82', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"random string","url":"http://www.google.com","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case83', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"james@bob.edu","url":"","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case84', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"james@bob.edu","url":"random string","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case85', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"james@bob.edu","url":"random string","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case86', async () => {
    const params = {"optStr":"random string","str":"","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case87', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"","url":"","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case88', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"","url":"random string","numFloat":-1000.5,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case89', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"","url":"random string","numFloat":2000345.7898,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case90', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"","url":"http://www.google.com","numFloat":0,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case91', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"random string","url":"","numFloat":-1000.5,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case92', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"random string","url":"","numFloat":2000345.7898,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case93', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"random string","url":"random string","numFloat":0,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case94', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"random string","url":"http://www.google.com","numFloat":0,"numInt":-1000};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case95', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"james@bob.edu","url":"","numFloat":-1000.5,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case96', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"james@bob.edu","url":"","numFloat":2000345.7898,"numInt":0};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case97', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"james@bob.edu","url":"random string","numFloat":0,"numInt":1};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });



  it('case98', async () => {
    const params = {"optStr":"random string","str":"random string","addressEmail":"james@bob.edu","url":"http://www.google.com","numFloat":-1000.5,"numInt":2004560};
    const result = await getSomething({}, params, undefined, mockGraphQLResolveInfo)
    expect(result).toMatchSnapshot();
  });


});
"
`;
